### PR TITLE
Add a select component

### PR DESCRIPTION
**Pull request by clemensp**

### CODE REVIEW

File: components/src/Form/Form.story.js [hunk 1]
```diff
@@ -122,6 +129,11 @@ storiesOf("Form", module)
           >
             <Input />
           </Field>
+          <Field
+            labelText="Project status"
```
Comment (coostenbrug) [inline components/src/Form/Form.story.js]: can move this prop to 1 line

File: components/src/Select/Select.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+const getBorderColor = (errored, isOpen, disabled, isFocused) => {
+  if (errored) { return theme.colors.red; }
+  if (disabled) { return theme.colors.lightGrey; }
+  if (isOpen || isFocused) { return theme.colors.blue; }
+
+  return theme.colors.grey;
+};
+
+const SelectBox = styled.div([], ({ error, isOpen, disabled }) => ({
```
Comment (coostenbrug) [inline components/src/Select/Select.js]: I assume eslint will catch this, but unused input args here

**Comment by coostenbrug** [comment]
Bugs:
- z-index of the menu should probably be increased so that it is always on top of other components 
- currently behaviour of if the menu is open and there are two select's is inconsistent: if top is open and bottom is closed, clicking bottom will close top. If top is open and bottom is closed, and bottom is opened, top remains open. EDIT: further investigation reveals this may be something related to another bug with defaultIsOpen. It seems like defaultIsOpen does not close the menu when an option is selected (both through clicking or with tab+enter).

File: components/src/Select/Select.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+          <SelectBox { ...getToggleButtonProps({ disabled, error, isOpen }) }>
+            <Input
+              { ...getInputProps({ disabled, error, isOpen }) } aria-required={ required }
+              value={ selectedItem && selectedItem.label } placeholder={ placeholder }
+              readOnly
+            />
+            <ToggleButton isOpen={ isOpen } />
+          </SelectBox>
+          {
+            isOpen
```
Comment (coostenbrug) [inline components/src/Select/Select.js]: consider using && for conditional rendering instead of turnery (looks nicer IMO, but others may disagree!)

Comment (clemensp) [reply components/src/Select/Select.js]: Fixed.

File: components/src/Form/Form.story.js (hunk 1 above)
Comment (clemensp) [reply components/src/Form/Form.story.js]: Fixed.

File: components/src/Select/Select.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+              && (
+                <Menu { ...getMenuProps({ error }) }>
+                  {
+                    isOpen
```
Comment (coostenbrug) [inline components/src/Select/Select.js]: You can probably drop this inner conditional render, the outer one is already based on isOpen anyway

**Pull request merged**